packages/flutter/test/material/debug_test.dart

```diff
@@ -10,7 +11,170 @@ void main() {
+      '   The ancestors of this widget were:\n'
+      '     [root]\n'
+    );
+    expect(error.diagnostics.length, 5);
+    expect(error.diagnostics[2].level, DiagnosticLevel.hint);
+    expect(
+      error.diagnostics[2].toStringDeep(),
+      equalsIgnoringHashCodes(
+        'To introduce a Material widget, you can either directly include\n'
+        'one, or use a widget that contains Material itself, such as a\n'
+        'Card, Dialog, Drawer, or Scaffold.\n',
+      ),
+    );
+    expect(error.diagnostics[3], isInstanceOf<DiagnosticsProperty<Element>>());
+    expect(error.diagnostics[4], isInstanceOf<DiagnosticsBlock>());
+  });
+
+  testWidgets('debugCheckHasMaterialLocalizations control test', (
+      WidgetTester tester) async {
+    await tester.pumpWidget(const BackButton());
+    final dynamic exception = tester.takeException();
+    expect(exception, isFlutterError);
+    final FlutterError error = exception;
+    expect(error.toStringDeep(),
+      'FlutterError\n'
+      '   No MaterialLocalizations found.\n'
+      '   BackButton widgets require MaterialLocalizations to be provided\n'
+      '   by a Localizations widget ancestor.\n'
+      '   Localizations are used to generate many different messages,\n'
+      '   labels,and abbreviations which are used by the material library.\n'
+      '   To introduce a MaterialLocalizations, either use a  MaterialApp\n'
+      '   at the root of your application to include them automatically, or\n'
+      '   add a Localization widget with a MaterialLocalizations delegate.\n'
+      '   The specific widget that could not find a MaterialLocalizations\n'
+      '   ancestor was:\n'
+      '     BackButton\n'
+      '   The ancestors of this widget were:\n'
+      '     [root]\n'
+    );
+    expect(error.diagnostics.length, 6);
+    expect(error.diagnostics[3].level, DiagnosticLevel.hint);
+    expect(
+      error.diagnostics[3].toStringDeep(),
+      equalsIgnoringHashCodes(
+        'To introduce a MaterialLocalizations, either use a  MaterialApp\n'
+        'at the root of your application to include them automatically, or\n'
+        'add a Localization widget with a MaterialLocalizations delegate.\n',
+      ),
+    );
+    expect(error.diagnostics[4], isInstanceOf<DiagnosticsProperty<Element>>());
+    expect(error.diagnostics[5], isInstanceOf<DiagnosticsBlock>());
+  });
+
+  testWidgets(
+      'debugCheckHasScaffold control test', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Builder(
+          builder: (BuildContext context) {
+            showBottomSheet<void>(context: context,
+                builder: (BuildContext context) => Container());
+            return Container();
+          }
+        ),
+      ),
+    );
+    final dynamic exception = tester.takeException();
+    expect(exception, isFlutterError);
+    final FlutterError error = exception;
+    expect(error.toStringDeep(), equalsIgnoringHashCodes(
```

Comment:
move the toStringDeep tests for the entire errors to be the very last test in the test block. That way if a test fails, the person reviewing the failure will know that all the other checks have passed and the only difference was a slightly probably harmless delta in the toStringDeep. For example, if there was a refactor and one ancestor changed then this test could start failing at some point for a harmless reason.

Reply:
This has just happened to me. Would the fix be to simply update the test with the latest error string? Or is it better to write this test in a different manner?

Reply:
@shihaohong  I wrote this test with the first fix that you mentioned in mind if there's any change made to the code. I figured that was the best way to test it since I had to write lot of test cases and I didn't have enough time to test it in a different manner. But if you feel like it's better to test in a different manner, please feel free to do so!
cmiiw? @jacob314